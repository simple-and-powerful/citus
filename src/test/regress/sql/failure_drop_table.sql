SET citus.next_shard_id TO 100300;
ALTER SEQUENCE pg_catalog.pg_dist_placement_placementid_seq RESTART 40;

SELECT citus.mitmproxy('flow.kill()');
SELECT citus.mitmproxy('recorder.reset()');

-- none of these commands generate network traffic
SET citus.shard_replication_factor TO 2;
CREATE TABLE append (id int);
SELECT create_distributed_table('append', 'id', 'append');
DROP TABLE append;

SELECT citus.mitmproxy('recorder.dump()');

SELECT citus.mitmproxy('flow.allow()');
CREATE TABLE append (id int);
SELECT create_distributed_table('append', 'id', 'append');
SELECT master_create_empty_shard('append');

SELECT citus.mitmproxy('recorder.dump()');

SELECT * FROM pg_dist_shard;
SELECT * FROM pg_dist_shard_placement;
-- drop the connection right at the beginning
SELECT citus.mitmproxy('flow.contains(b"assign_distributed_transaction_id").kill()');
DROP TABLE append;
SELECT * FROM pg_dist_shard;
SELECT * FROM pg_dist_shard_placement;

SELECT citus.mitmproxy('flow.allow()');
CREATE TABLE append (id int);
SELECT create_distributed_table('append', 'id', 'append');
SELECT master_create_empty_shard('append');

SELECT * FROM pg_dist_shard;
SELECT * FROM pg_dist_shard_placement;
SELECT citus.mitmproxy('flow.contains(b"DROP TABLE IF EXISTS").kill()');
DROP TABLE append;
SELECT * FROM pg_dist_shard;
SELECT * FROM pg_dist_shard_placement;

-- next steps:
-- PREPARE TRANSACTION
-- COMMIT PREPARED
