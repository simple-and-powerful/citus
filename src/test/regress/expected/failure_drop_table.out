SET citus.next_shard_id TO 100300;
ALTER SEQUENCE pg_catalog.pg_dist_placement_placementid_seq RESTART 40;
SELECT citus.mitmproxy('flow.kill()');
 mitmproxy 
-----------
 
(1 row)

SELECT citus.mitmproxy('recorder.reset()');
 mitmproxy 
-----------
 
(1 row)

-- none of these commands generate network traffic
SET citus.shard_replication_factor TO 2;
CREATE TABLE append (id int);
SELECT create_distributed_table('append', 'id', 'append');
 create_distributed_table 
--------------------------
 
(1 row)

DROP TABLE append;
SELECT citus.mitmproxy('recorder.dump()');
 mitmproxy 
-----------
 
(1 row)

SELECT citus.mitmproxy('flow.allow()');
 mitmproxy 
-----------
 
(1 row)

CREATE TABLE append (id int);
SELECT create_distributed_table('append', 'id', 'append');
 create_distributed_table 
--------------------------
 
(1 row)

SELECT master_create_empty_shard('append');
 master_create_empty_shard 
---------------------------
                    100300
(1 row)

SELECT citus.mitmproxy('recorder.dump()');
                                                                                                                                                                                                                                                                                                         mitmproxy                                                                                                                                                                                                                                                                                                          
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 
 message from client: \x00\x00\x00W\x00\x03\x00\x00user\x00postgres\x00database\x00regression\x00application_name\x00citus\x00client_encoding\x00UTF8\x00\x00
 message from server: R\x00\x00\x00\x08\x00\x00\x00\x00S\x00\x00\x00\x1bapplication_name\x00citus\x00S\x00\x00\x00\x19client_encoding\x00UTF8\x00S\x00\x00\x00\x17DateStyle\x00ISO, MDY\x00S\x00\x00\x00\x19integer_datetimes\x00on\x00S\x00\x00\x00\x1bIntervalStyle\x00postgres\x00S\x00\x00\x00\x14is_superuser\x00on\x00S\x00\x00\x00\x19server_encoding\x00UTF8\x00S\x00\x00\x00\x18server_version\x0010.3\x00S\x00\x00\x00#session_authorization\x00postgres\x00S\x00\x00\x00#standard_conforming_strings\x00on\x00S\x00\x00\x00\x18TimeZone\x00US/Pacific\x00K\x00\x00\x00\x0c\x00\x00o=\xfbbU\x81Z\x00\x00\x00\x05I
 message from client: Q\x00\x00\x00^SELECT worker_apply_shard_ddl_command (100300, 'CREATE TABLE public.append (id integer)')\x00
 message from server: T\x00\x00\x007\x00\x01worker_apply_shard_ddl_command\x00\x00\x00\x00\x00\x00\x00\x00\x00\x08\xe6\x00\x04\xff\xff\xff\xff\x00\x00D\x00\x00\x00\n\x00\x01\x00\x00\x00\x00C\x00\x00\x00\rSELECT 1\x00Z\x00\x00\x00\x05I
 message from client: X\x00\x00\x00\x04
(6 rows)

SELECT * FROM pg_dist_shard;
 logicalrelid | shardid | shardstorage | shardminvalue | shardmaxvalue 
--------------+---------+--------------+---------------+---------------
 append       |  100300 | t            |               | 
(1 row)

SELECT * FROM pg_dist_shard_placement;
 shardid | shardstate | shardlength | nodename  | nodeport | placementid 
---------+------------+-------------+-----------+----------+-------------
  100300 |          1 |           0 | localhost |    57637 |          40
  100300 |          1 |           0 | localhost |    57640 |          41
(2 rows)

-- drop the connection right at the beginning
SELECT citus.mitmproxy('flow.contains(b"assign_distributed_transaction_id").kill()');
 mitmproxy 
-----------
 
(1 row)

DROP TABLE append;
WARNING:  connection not open
CONTEXT:  while executing command on localhost:57640
SQL statement "SELECT master_drop_all_shards(v_obj.objid, v_obj.schema_name, v_obj.object_name)"
PL/pgSQL function citus_drop_trigger() line 17 at PERFORM
WARNING:  could not connect to shard "append_100300" on node "localhost:57640"
DETAIL:  Marking this shard placement for deletion
CONTEXT:  SQL statement "SELECT master_drop_all_shards(v_obj.objid, v_obj.schema_name, v_obj.object_name)"
PL/pgSQL function citus_drop_trigger() line 17 at PERFORM
WARNING:  connection not open
CONTEXT:  while executing command on localhost:57640
WARNING:  connection not open
CONTEXT:  while executing command on localhost:57640
WARNING:  connection not open
CONTEXT:  while executing command on localhost:57640
SELECT * FROM pg_dist_shard;
 logicalrelid | shardid | shardstorage | shardminvalue | shardmaxvalue 
--------------+---------+--------------+---------------+---------------
(0 rows)

SELECT * FROM pg_dist_shard_placement;
 shardid | shardstate | shardlength | nodename  | nodeport | placementid 
---------+------------+-------------+-----------+----------+-------------
  100300 |          3 |           0 | localhost |    57640 |          41
(1 row)

SELECT citus.mitmproxy('flow.allow()');
 mitmproxy 
-----------
 
(1 row)

CREATE TABLE append (id int);
SELECT create_distributed_table('append', 'id', 'append');
 create_distributed_table 
--------------------------
 
(1 row)

SELECT master_create_empty_shard('append');
 master_create_empty_shard 
---------------------------
                    100301
(1 row)

SELECT * FROM pg_dist_shard;
 logicalrelid | shardid | shardstorage | shardminvalue | shardmaxvalue 
--------------+---------+--------------+---------------+---------------
 append       |  100301 | t            |               | 
(1 row)

SELECT * FROM pg_dist_shard_placement;
 shardid | shardstate | shardlength | nodename  | nodeport | placementid 
---------+------------+-------------+-----------+----------+-------------
  100301 |          1 |           0 | localhost |    57637 |          43
  100300 |          3 |           0 | localhost |    57640 |          41
  100301 |          1 |           0 | localhost |    57640 |          42
(3 rows)

SELECT citus.mitmproxy('flow.contains(b"DROP TABLE IF EXISTS").kill()');
 mitmproxy 
-----------
 
(1 row)

DROP TABLE append;
ERROR:  server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
CONTEXT:  while executing command on localhost:57640
SQL statement "SELECT master_drop_all_shards(v_obj.objid, v_obj.schema_name, v_obj.object_name)"
PL/pgSQL function citus_drop_trigger() line 17 at PERFORM
SELECT * FROM pg_dist_shard;
 logicalrelid | shardid | shardstorage | shardminvalue | shardmaxvalue 
--------------+---------+--------------+---------------+---------------
 append       |  100301 | t            |               | 
(1 row)

SELECT * FROM pg_dist_shard_placement;
 shardid | shardstate | shardlength | nodename  | nodeport | placementid 
---------+------------+-------------+-----------+----------+-------------
  100301 |          1 |           0 | localhost |    57637 |          43
  100300 |          3 |           0 | localhost |    57640 |          41
  100301 |          1 |           0 | localhost |    57640 |          42
(3 rows)

-- next steps:
-- PREPARE TRANSACTION
-- COMMIT PREPARED
